### DISCUSSION TITLE
Named pipe connect failed after more than 1000 connections

### DISCUSSION THREAD

**Discussion by willjiang1024** [libuv/libuv#3310] [Q&A]
Hi, all

I tested the benchmark case pipe_pound_1000 with below modifications but get a failed result.
`$ git diff test/benchmark-pound.c
diff --git a/test/benchmark-pound.c b/test/benchmark-pound.c
index 830bc554..09e55a26 100644
--- a/test/benchmark-pound.c
+++ b/test/benchmark-pound.c
@@ -23,7 +23,7 @@
 #include "uv.h"

 /* Update this is you're going to run > 1000 concurrent requests. */
-#define MAX_CONNS 1000
+#define MAX_CONNS 2000

 #undef NANOSEC
 #define NANOSEC ((uint64_t) 1e9)
@@ -342,7 +342,7 @@ BENCHMARK_IMPL(pipe_pound_100) {

 BENCHMARK_IMPL(pipe_pound_1000) {
-  return pound_it(1000,
+  return pound_it(1020,
                   "pipe",
                   pipe_do_setup,
                   pipe_do_connect,
`

The test result is like below:
`
$ ./uv_run_benchmarks_a pipe_pound_1000
ok 1 - pipe_pound_1000
# pipe-conn-pound-1020: 128682 accepts/s (20056 failed)
`

 I tested it in Ubuntu 20.04.1 LTS, and libuv is v1.x. Has anyone seen this issue before? Is there any limitation on the number of pipe connections in Linux?

Thanks

**Comment by willjiang1024** [comment]
When i enabled the DEBUG, i got an error message like "connect error EMFILE", it seems a limitation of the process opened too many files. Am i right?

**Comment by bnoordhuis** [comment]
Yes. There's a default 1,024 file descriptor limit on Ubuntu (and most - but not all - other distros.)

You can increase it with `ulimit -n`.
